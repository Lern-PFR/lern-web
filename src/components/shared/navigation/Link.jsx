import PropTypes from 'prop-types';
import { Link as ReactRouterLink } from 'react-router-dom';
import styled from 'styled-components';
import { color } from 'styled-system';

const StyledLink = styled(ReactRouterLink)(
	{
		cursor: 'inherit',
		color: 'inherit',
		fontFamily: 'inherit',
		fontSize: 'inherit',
		fontWeight: 'inherit',
		textDecoration: 'inherit',
	},
	color,
);

/**
 * @name Link
 * @description Renders a link to an internal URL, by providing keys path to the route.
 *
 * @author Yann Hodiesne
 *
 * @param {string}	to			key of the route to target with this Link
 * @param {bool}	[disabled]	Whether the link is disabled or not. If so, the onClick event is aborted.
 */
const Link = ({ to, disabled, ...props }) => (
	<StyledLink to={to} {...(disabled ? { onClick: (e) => e.preventDefault() } : {})} {...props} />
);

Link.propTypes = {
	to: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.object,
		PropTypes.func,
	]),
	disabled: PropTypes.bool,
};

Link.defaultProps = {
	disabled: false,
};

Link.defaultProps = {
	to: '',
};

export default Link;
