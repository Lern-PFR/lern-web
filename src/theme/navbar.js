import { coconut, oyster } from './colors';
import { pica } from './textStyles';

const navbar = {
	display: 'grid',
	gridTemplateColumns: 'max-content 1fr max-content',
	width: '100%',
	height: '80px',
	gridGap: '3em',
	alignItems: 'center',
	paddingX: '3em',
	backgroundColor: coconut.default,
	// @TODO: style shadow / blur
};

const navbarLogo = {
	cursor: 'pointer',
	height: '2em',
};

const navbarLinkList = {
	listStyle: 'none',
	display: 'flex',
	flexDirection: 'row',
	gridGap: '1em',
	padding: 0,
};

const navbarLink = {
	cursor: 'pointer',
	textAlign: 'center',
	transition: 'font-weight .1s ease-out',
	color: oyster.default,
	...pica,

	'&:hover': {
		fontWeight: 600,
	},
	// @TODO: style active link.
};

export {
	navbar,
	navbarLogo,
	navbarLinkList,
	navbarLink,
};
