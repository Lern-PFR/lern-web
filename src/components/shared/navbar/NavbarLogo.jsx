import { ReactSVG } from 'react-svg';
import styled from 'styled-components';
import conf from 'conf';
import { navbarLogo } from 'theme/navbar';

const StyledSvg = styled(ReactSVG)(
	{
		'& svg': {
			...navbarLogo,
		},
	}
);

/**
 * @name NavbarLogo
 * @description The svg file to display in the main navbar component.
 *
 * @author TimothÃ©e Simon-Franza
 */
const NavbarLogo = () => (
	<StyledSvg src={`${conf.svgPath}/logo_black_mini.svg`} />
);

export default NavbarLogo;
