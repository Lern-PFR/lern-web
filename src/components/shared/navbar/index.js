import Navbar from './Navbar';
import navbarLink from './NavbarLink';

export {
	Navbar as default,
	navbarLink,
};
