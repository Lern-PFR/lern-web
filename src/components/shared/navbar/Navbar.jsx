import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import { navbar, navbarLinkList } from 'theme/navbar';
import { StyledDiv, StyledList } from 'components/shared/styledElements';
import NavbarLink from './NavbarLink';
import NavbarLogo from './NavbarLogo';
// @TODO: Add routing.
/**
 * @name Navbar
 * @description The main navbar component.
 *
 * @author TimothÃ©e Simon-Franza
 *
 * @param {func} t The translation method provided by the withTranslation HoC.
 */
const Navbar = ({ t }) => (
	<StyledDiv as="nav" {...navbar}>
		<NavbarLogo />
		<StyledList {...navbarLinkList}>
			<NavbarLink targetUrl="">{t('navigation.menu.home')}</NavbarLink>
			<NavbarLink targetUrl="">{t('navigation.menu.my_courses')}</NavbarLink>
			<NavbarLink targetUrl="">{t('navigation.menu.about')}</NavbarLink>
			<NavbarLink targetUrl="">{t('navigation.menu.contact_us')}</NavbarLink>
		</StyledList>
	</StyledDiv>
);

Navbar.propTypes = {
	t: PropTypes.func.isRequired,
};

export default withTranslation()(Navbar);
