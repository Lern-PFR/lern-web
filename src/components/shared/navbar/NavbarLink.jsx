import PropTypes from 'prop-types';
import { Link } from 'components/shared/navigation';
import { StyledListItem } from 'components/shared/styledElements';
import { navbarLink } from 'theme/navbar';
import styled from 'styled-components';

const StyledNavbarLinkContainer = styled(StyledListItem)(
	{
		...navbarLink,
		'& > * > *::first-letter': {
			textTransform: 'uppercase',
		},
		'& > * > *:not(::first-letter)': {
			textTransform: 'lowercase',
		},
	}
);

/**
 * @name NavbarLink
 * @description The link to display inside the Navbar component.
 *
 * @author TimothÃ©e Simon-Franza
 *
 * @param {string}	children	The text to display.
 * @param {string}	targetUrl	The url to redirect to on click.
 */
const NavbarLink = ({ children, targetUrl }) => (
	<StyledNavbarLinkContainer {...navbarLink}>
		<Link to={targetUrl}>{children}</Link>
	</StyledNavbarLinkContainer>
);

NavbarLink.propTypes = {
	children: PropTypes.string.isRequired,
	targetUrl: PropTypes.string.isRequired,
};

export default NavbarLink;
