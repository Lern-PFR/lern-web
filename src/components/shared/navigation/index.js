import Link from './Link';
import Navbar from './navbar';

export {
	Link,
	Navbar,
};
