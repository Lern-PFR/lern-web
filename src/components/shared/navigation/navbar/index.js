import Navbar from './Navbar';
import NavbarLink from './NavbarLink';
import NavbarLogo from './NavbarLogo';

export {
	Navbar as default,
	NavbarLink,
	NavbarLogo,
};
