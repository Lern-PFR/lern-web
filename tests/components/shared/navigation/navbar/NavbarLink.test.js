import { shallow } from 'enzyme';
import { NavbarLink } from 'components/shared/navigation/navbar';

describe('NavbarLink', () => {
	it('should match previous snapshot', () => {
		const sut = (<NavbarLink targetUrl="">test</NavbarLink>);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});
});
