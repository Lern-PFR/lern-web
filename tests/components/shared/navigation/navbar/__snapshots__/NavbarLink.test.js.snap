// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLink should match previous snapshot 1`] = `
<Styled(styled.li)
  &:hover={
    Object {
      "fontWeight": 600,
    }
  }
  color="#403B3B"
  cursor="pointer"
  fontFamily="IBM Plex Sans, Helvetica Neue, Arial, sans-serif"
  fontSize="1rem"
  fontWeight={400}
  letterSpacing={0.16}
  lineHeight="1.5em"
  textAlign="center"
  transition="font-weight .1s ease-out"
>
  <Link
    to=""
  >
    test
  </Link>
</Styled(styled.li)>
`;
